src/west/app/project.py

```diff
@@ -186,6 +186,9 @@ def do_add_parser(self, parser_adder):
         parser.add_argument('-m', '--manifest-url',
                             help='''manifest repository URL to clone;
                             cannot be combined with -l''')
+        parser.add_argument('-d', '--depth',
+                            help='''clone depth of repository;
+                            cannot be combined with -l''')
```

Comment:
I think we don't want to split all possible git options like this, instead we could do the same as the `update -o` command:

```python
        parser.add_argument('-o', '--clone-opt', action='append', default=[],
                            help='''additional option to pass to 'git clone'
                            (e.g. 'o=--depth=1');
                            may be given more than once;
                            cannot be combined with -l''')
```